Refactor toolkit.nu

texastoland opened the pull request:
Closes #789 🍻

- Runs without first generating a script
- Returns error (file) count
- Requires `--and-exit` to exit with error code
- Enables alternative report with env `STUB_IDE_CHECK=true`
- Expands documentation
- All subcommands share same file querying
- Prepares for nupm test integration

texastoland commented:
@AucaCoyan If you wanna review 📝

Tested combinations of every flag, every subcommand, as well as the stub locally. Not able to repro the CI failure yet. Debugging now.

AucaCoyan commented on toolkit.nu:
```diff
@@ -4,61 +4,102 @@
+    let file = $in
+    let stub = $env.STUB_IDE_CHECK? | default false | into bool
+    let diagnostics = if $stub {
+        do { nu -n -c $"use '($file)'" }
```
I like to use full flags in commands, just in case in the future some of them get deprecated, I have to lookup what `-c` meant in the first place
`nu -no-config-file --commands`

AucaCoyan reviewed (changes requested):
Excellent additions! I'm happier with this code that whatever I wrote in the first place ❤️ 
Can I ask for you for another change?
on `.github/worklows/daily` you need to change LN 29 to
`nu --commands "use toolkit.nu *; check pr --full --and-exit"`
And also remove the last step

I tested this on my repo and it worked as expected (fixes #789 )
You can see the ❌ in the [main branch](https://github.com/AucaCoyan/nu_scripts/tree/main) and also [the results in the CI](https://github.com/AucaCoyan/nu_scripts/actions/runs/8279935393/job/22655376091)

AucaCoyan commented:
fixes #789

texastoland commented:
> on `.github/worklows/daily` you need to change LN 29 to `nu --commands "use toolkit.nu *; check pr --full --and-exit"` And also remove the last step

Great catch and fixed! The unrelated YAML changes were just Prettier.

> closes #789

Added to the description!

@fdncred I'm planning to reorganize this into a directory module when I integrate `nupm test`. I have to add nupm to hustcer/setup-nu/issues/40 first.

AucaCoyan reviewed (approved):
All good!

fdncred commented:
I'm not a huge fan of using nupm before it's ready because it may result in endless changes here.

texastoland commented:
> I'm not a huge fan of using nupm before it's ready because it may result in endless changes here.

Only for `nupm test` since the other is deprecated?

fdncred commented:
Another reason why I think std testing was removed too soon. If we paint ourselves into a corner here with nupm, we can always remove this functionality from the ci.

texastoland commented:
I don't mind deferring it. Provided details on Discord. Simplifying the GitHub action (just broke it) then this will be ready to go 👌🏼

fdncred commented:
I'm fine with trying things out and seeing how they go. No worries. I won't block this.

texastoland commented:
> I don't mind deferring it. Provided details on Discord. I'm simplifying the GitHub action (just broke it) then this will be ready to go 👌🏼

@fdncred :shipit:

kubouch commented:
The `nupm test` is continually tested in nupm's CI, so it should work OK, but its implementation might change, and then the tests would need updating. But it might be worth having nupm being used somewhere for testing and nu_scripts is not so critical, so we could try it and see how it goes.

texastoland commented:
> @fdncred: I'm fine with trying things out and seeing how they go. No worries. I won't block this.

To clarify there's no testing yet.

> [me]: I have to add nupm to [hustcer/setup-nu/issues/40](https://github.com/hustcer/setup-nu/issues/40) first.

This is prerequisite. I'm already attempting it!

> @kubouch: nu_scripts is not so critical, so we could try it and see how it goes.

I'll tag you in the next PR then ✌🏼
